Fix up relevant CA1416 warnings - Set 3

Tamilarasan-Paranthaman opened the pull request:
### Description of Change

The CA1416 warnings were already resolved in PR https://github.com/dotnet/maui/pull/26593  

In this PR, we have divided the file changes from the previous PR into smaller, more manageable groups and added test cases for the affected areas to ensure proper functionality.

Tamilarasan-Paranthaman committed:
CA1416 warning fix

Redth commented:
/azp run

Tamilarasan-Paranthaman committed:
Added snapshots

PureWeen commented:
/azp run

mattleibow commented:
/rebase

mattleibow commented on src/Essentials/src/Screenshot/Screenshot.ios.cs:
```diff
@@ -31,66 +31,104 @@ public Task<IScreenshotResult> CaptureAsync(UIWindow window)
-
-			// ctx will be null if the width/height of the view is zero
-			if (ctx is not null && !TryRender(window, out _))
+			if (!OperatingSystem.IsIOSVersionAtLeast(17))
```
nit: can we reverse the polarity here :) 

It makes it a bit easier to read if the first is the latest version and then the else gets older. If 17 else 16 else 15.

mattleibow replied:
Also, why do we need this block for ios 17+? It seems to have been around since iOS 10? Can we just use that code? The UIGraphics style is obsolete in iOS 18, but we don't need to use that as the main flag here.

Tamilarasan-Paranthaman replied:
@mattleibow, it seems that BeginImageContextWithOptions is not supported on iOS 17.0 and later.

<img width="673" alt="image" src="https://github.com/user-attachments/assets/a56ad8eb-4efa-4a99-9c77-7ed4286b0df7" />


<img width="527" alt="image" src="https://github.com/user-attachments/assets/84dbf502-83ce-4022-9b6e-97c7ca90e9be" />




As @mattleibow suggested [here](https://github.com/dotnet/maui/pull/26751#discussion_r1904349779), should we adopt the new approach for older versions as well, @PureWeen?
 
```C#

public Task<IScreenshotResult> CaptureAsync(UIWindow window)
{
	_ = window ?? throw new ArgumentNullException(nameof(window));

	using var renderer = new UIGraphicsImageRenderer(window.Bounds.Size);
	var resultImage = renderer.CreateImage((UIGraphicsImageRendererContext imageContext) =>
	{
		if (!TryRender(window, out _))
		{
			// TODO: test/handle this case
		}
	});
	var screenshotResult = new ScreenshotResult(resultImage);
	return Task.FromResult<IScreenshotResult>(screenshotResult);
}

var screenshotResult = new ScreenshotResult(resultImage);
return Task.FromResult<IScreenshotResult?>(screenshotResult);
```

Could you please share your thoughts on this, @PureWeen ?

mattleibow replied:
Can we just replace and use the new method?

mattleibow commented on src/Core/src/Platform/iOS/MauiUIApplicationDelegate.Menu.cs:
```diff
@@ -29,13 +29,15 @@ public override void BuildMenu(IUIMenuBuilder builder)
 
 			UIWindow? window = null;
-			if (OperatingSystem.IsMacCatalystVersionAtLeast(14))
+
+			// WindowingBehaviors is only available on iOS 16+ and Mac Catalyst
+			if (OperatingSystem.IsMacCatalystVersionAtLeast(16) || OperatingSystem.IsIOSVersionAtLeast(16))
```
I think this changes the logic? We need to keep the iOS 14 here, but wrap the inner if block on lines 51/53 with this check. We still need to have this block for the fallback on line 66 which is in this main block.

mattleibow commented on src/Core/src/ImageSources/iOS/ImageSourceExtensions.cs:
```diff
@@ -35,20 +35,18 @@ public static partial class ImageSourceExtensions
-			UIGraphics.EndImageContext();
-
-			return image.ImageWithRenderingMode(UIImageRenderingMode.Automatic);
+			var renderer = new UIGraphicsImageRenderer(imagesize);
```
For example, here you switched directly to the new way without keeping the old way. I think this is better? Thoughts?

mattleibow reviewed (changes requested):
Thanks for the PR, I have a review and I think we can just move to the new things. No need to have old code in here that we have to maintain just because we always had it.

And one small if block change needed.

Tamilarasan-Paranthaman committed:
Reverting files changes

mattleibow commented:
/azp run

mattleibow commented on src/Essentials/src/Screenshot/Screenshot.ios.cs:
```diff
@@ -31,66 +31,104 @@ public Task<IScreenshotResult> CaptureAsync(UIWindow window)
-
-			// ctx will be null if the width/height of the view is zero
-			if (ctx is not null && !TryRender(view, out _))
+			if (!OperatingSystem.IsIOSVersionAtLeast(17))
```
Can we use the new method here instead of the different mechanisms.

Tamilarasan-Paranthaman replied:
@mattleibow, I have removed the old approach.

mattleibow commented on src/Essentials/src/Screenshot/Screenshot.ios.cs:
```diff
@@ -31,66 +31,104 @@ public Task<IScreenshotResult> CaptureAsync(UIWindow window)
+			}
 
-			return Task.FromResult<IScreenshotResult?>(result);
+			using var renderer = new UIGraphicsImageRenderer(layer.Bounds.Size);
```
Just use the new way.

Tamilarasan-Paranthaman replied:
Modified

mattleibow commented on src/Graphics/src/Graphics/Platforms/iOS/UIImageExtensions.cs:
```diff
@@ -27,37 +29,75 @@ public static UIImage ScaleImage(this UIImage target, float maxWidth, float maxH
-			UIGraphics.EndImageContext();
-
-			if (disposeOriginal)
+			if (!OperatingSystem.IsIOSVersionAtLeast(17))
```
Just use the new way for both chunks here.

Tamilarasan-Paranthaman replied:
I have modified it

mattleibow reviewed (changes requested):
I think we should just use the new mechanism for drawing and not have the old way and checks.

Tamilarasan-Paranthaman committed:
Removed the old approach

Tamilarasan-Paranthaman committed:
Revert unwanted changes

rmarinho commented:
Can we please rebase maybe target the inflight branch is better

NafeelaNazhir committed:
[Testing] Feature Matrix UITest Cases for CollectionView Dynamic changes (#29424)

* Added FeatureTests for CollectionView

* changes updated

* changes updated

Tamilarasan-Paranthaman committed:
Reverted unwanted changes

Tamilarasan-Paranthaman committed:
Reverted unwanted changes

github-actions[bot] committed:
[create-pull-request] automated change (#29505)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

Tamilarasan-Paranthaman commented:
> Can we please rebase maybe target the inflight branch is better

@rmarinho, I have rebased the branch, resolved the conflicts, and updated the target to the `inflight/current`  branch. Please have a look and let me know if you have any concerns.

mattleibow reviewed (changes requested):
I think something went wrong with the rebase/merge... :(

mattleibow commented:
@PureWeen you broke it! 😿

Tamilarasan-Paranthaman committed:
Revert unwanted changes

Tamilarasan-Paranthaman commented:
> I think something went wrong with the rebase/merge... :(

@mattleibow  I have rebased the branch and resolved the conflicts. Please have a look and let me know if you have any concerns.
